lib/ansible/utils/path.py

```diff
@@ -103,3 +104,18 @@ def basedir(source):
         dname = os.path.abspath(dname)
 
     return to_text(dname, errors='surrogate_or_strict')
+
+
+def cleanup_tmp_dir(path):
+    """Removes temporary directory"""
+    try:
+        if os.path.isdir(path):
+            try:
+                shutil.rmtree(path)
+            except Exception as e:
+                # Importing here to avoid circular import
+                from ansible.utils.display import Display
+                display = Display()
+                display.display('Unable to remove temporary file {0}'.format(to_native(e)))
```

Comment:
Anything going through Display() needs to use to_text()

Reply:
And you should then mark the literal string as a u"" literal:
```
                 display.display(u'Unable to remove temporary file {0}'.format(to_text(e)))
```

Reply:
Thank you. I fixed it.